### PR TITLE
Javascript week1

### PR DESCRIPTION
**Pull request by l0rians**
Homework: 

1. Age-Ify
2. Goodboy - Oldboy
3. Housey Pricey
4. Ez Namey

### CODE REVIEW

File: Javascript/Age-ify (A future age calculator)/age.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+let yearOfBirth = 1999;
+let yearFuture = 2049;
+
+let age = yearFuture - yearOfBirth;
```
Comment (jason-vasilev) [inline Javascript/Age-ify (A future age calculator)/age.js]: You can set those variables with a `const`. Use `let` only when you are to reassign their values.

File: Javascript/Ez Namey /namey.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+let firstWords = [
```
Comment (jason-vasilev) [inline Javascript/Ez Namey /namey.js]: Nearly all `let` in this file can be set as `const`.

File: Javascript/Ez Namey /namey.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  "Creations",
+  "Inventions",
+  "Platforms",
+  "Ventures",
+  "Projects",
+];
+
+const randomNumber = Math.floor(Math.random() * 10);
+
+let startupName = firstWords[randomNumber] + " " + secondWords[randomNumber];
```
Comment (jason-vasilev) [inline Javascript/Ez Namey /namey.js]: Using the same `randomNumber` for both `firstWords` and `secondWords` will limit the amount of combinations you can get generated.  In a way you will always get "Amazing Solutions", "Creative Ideas" etc, but would never get "Amazing Ideas", or "Innovative Solutions".

File: Javascript/Ez Namey /namey.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+let nameLength = startupName.length;
```
Comment (jason-vasilev) [inline Javascript/Ez Namey /namey.js]: Use `const` until it doesn't `let` you to ;)

File: Javascript/Goodboy-Oldboy /script.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+let dogYearBirth = 2014;
+let dogYearFuture = 2031;
+let shouldShowResultInDogYears = true;
+
+let dogAge;
```
Comment (jason-vasilev) [inline Javascript/Goodboy-Oldboy /script.js]: It's a good practice to define variables as a number or empty string, if you already know what they would be changed to. In this case:
```suggestion
let dogAge = 0;
```

File: Javascript/Goodboy-Oldboy /script.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+if (shouldShowResultInDogYears) {
```
Comment (jason-vasilev) [inline Javascript/Goodboy-Oldboy /script.js]: You repeat this line of code twice. Why not combine the logic at one place?

File: Javascript/Goodboy-Oldboy /script.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  dogAge = dogYearFuture - dogYearBirth;
```
Comment (jason-vasilev) [inline Javascript/Goodboy-Oldboy /script.js]: This line is repeated in both cases - true and false. What if you create a variable outside the `if else` statement, and you write here directly the `console.log`s? Then you can avoid repeating `dogYearFuture - dogYearBirth` two times, and having two separate `if else` statements.

File: Javascript/Housey pricey/house.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+// Peter's parameters
+let peterWidth = 8;
```
Comment (jason-vasilev) [inline Javascript/Housey pricey/house.js]: Imagine, you don't have context of what the assignment is about and you read those variable names. They might not be the most clear.
`peterHouseWidth` would be more appropriate and descriptive. Also you can set them as `const`.

File: Javascript/Housey pricey/house.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+let peterHeight = 10;
+let peterDepth = 10;
+let peterGardenSize = 100;
+let peterRealPrice = 2500000;
+
+// Julia's parameters
+let juliaWidth = 5;
+let juliaHeigth = 8;
+let juliaDepth = 11;
+let juliaGardenSize = 70;
```
Comment (jason-vasilev) [inline Javascript/Housey pricey/house.js]: Well done using [`cammelCase`](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/#camel-case)

Comment (jason-vasilev) [inline Javascript/Housey pricey/house.js]: All those variables could also be set through an object. Then you can access the properties as `peterHouse.depth`,  `peterHouse.width` for example. Nothing wrong with what you've written - this is just another suggestion ( :

**Review by jason-vasilev — APPROVED** [review]
I've left you some suggestions for reducing repeated code, and for using `const` instead of `let`. You are welcome to push a commit with updates, or merge, as I am approving.

File: Javascript/Housey pricey/house.js (hunk 8 above)
Comment (l0rians) [reply Javascript/Housey pricey/house.js]: ****Changed.****

File: Javascript/Goodboy-Oldboy /script.js (hunk 7 above)
Comment (l0rians) [reply Javascript/Goodboy-Oldboy /script.js]: ****Changed.****

File: Javascript/Housey pricey/house.js (hunk 9 above)
Comment (l0rians) [reply Javascript/Housey pricey/house.js]: ***Thank you!***

Comment (l0rians) [reply Javascript/Housey pricey/house.js]: ****Thank you. Next time i will try to write better code!****

File: Javascript/Goodboy-Oldboy /script.js (hunk 6 above)
Comment (l0rians) [reply Javascript/Goodboy-Oldboy /script.js]: ***Mentor, you mean to use the ternary operator?***

File: Javascript/Ez Namey /namey.js (hunk 4 above)
Comment (l0rians) [reply Javascript/Ez Namey /namey.js]: ***Changed.***

File: Javascript/Ez Namey /namey.js (hunk 3 above)
Comment (l0rians) [reply Javascript/Ez Namey /namey.js]: ***Got it. Changed.***

File: Javascript/Ez Namey /namey.js (hunk 2 above)
Comment (l0rians) [reply Javascript/Ez Namey /namey.js]: ***Changed.***

File: Javascript/Age-ify (A future age calculator)/age.js (hunk 1 above)
Comment (l0rians) [reply Javascript/Age-ify (A future age calculator)/age.js]: ***Got it. Changed.***

File: Javascript/Goodboy-Oldboy /script.js (hunk 6 above)
Comment (jason-vasilev) [reply Javascript/Goodboy-Oldboy /script.js]: Screenshot 2024-06-05 at 5 19 05 PM
The highlighted is repeated. You could combine it into a single `if else`.

File: Javascript/Housey pricey/house.js (hunk 8 above)
Comment (jason-vasilev) [reply Javascript/Housey pricey/house.js]: I am glad you are getting things changed. It's a great way to go through the code and do even more learning.
It's not necessary to reply to every comment. Once you push a commit, I get an email notification. If you write a comment eg "Changed", "Fixed" or something else, I get a new email, but each time a new commit is pushed that makes changes to a particular line of code that I've commented on, I can see this "Outdated" label, which means that my comment has been addressed ;) This is just a heads-up :)
Screenshot 2024-06-05 at 5 20 04 PM

File: Javascript/Goodboy-Oldboy /script.js [hunk 10]
```diff
@@ -1,18 +1,19 @@
...
 if (shouldShowResultInDogYears) {
-  console.log(`Your dog will be ${dogAge} dog years old in ${dogYearFuture}`);
+  console.log(
+    `Your dog will be ${dogAgeInDogYears} dog years old in ${dogYearFuture}`
+  );
 } else {
-  console.log(`Your dog will be ${dogAge} human years old in ${dogYearFuture}`);
+  console.log(
+    `Your dog will be ${dogAgeInHumanYears} human years old in ${dogYearFuture}`
+  );
```
Comment (jason-vasilev) [inline Javascript/Goodboy-Oldboy /script.js]: No need to update it now, but as you use `dogAgeInDogYears` inside the `if else` you can definite it there, so if the condition is not true, you won't need to define it anyways.
```
const dogAgeInHumanYears = dogYearFuture - dogYearBirth;

if (shouldShowResultInDogYears) {
  const dogAgeInDogYears = dogAgeInHumanYears * 7;
  
  console.log(
    `Your dog will be ${dogAgeInDogYears} dog years old in ${dogYearFuture}`
  );
} else {
  console.log(
    `Your dog will be ${dogAgeInHumanYears} human years old in ${dogYearFuture}`
  );
}
```

**Pull request merged**